### PR TITLE
Message Queue and Priority Tests

### PR DESCRIPTION
**Pull request by coffee-cup**
## Changes

- Added a message queue so that message receiving is fair and FIFO
- Allow `wcet = 0`
- **Abort with timing violation if a periodic task does not start on ttns**
- Dispatch after task creation
- Added more message trace tests for testing FIFO
- Added task tests to check task priority is respected
- Add `lights.c`

## Closes

- https://github.com/coffee-cup/uvic-csc460/issues/18
- https://github.com/coffee-cup/uvic-csc460/issues/8
- https://github.com/coffee-cup/uvic-csc460/issues/16

### CODE REVIEW

**Comment by CanadianCommander** [comment]
good to see this repo is back ^^

File: project2/kernel.c [hunk 1]
```diff
@@ -217,6 +224,9 @@ void Kernel_Task_Create() {
         OS_Abort(NO_DEAD_PROCESS);
         return;
     }
+
+    /* Dispatch because created task might be higher priority then the currently running task */
+    Dispatch();
```
Comment (zevisert) [inline project2/kernel.c]: Move this to Kernel_Request_Create

File: project2/kernel.c [hunk 2]
```diff
@@ -103,6 +103,8 @@ extern void Enter_Kernel();
 /* User level 'main' function */
 extern void setup(void);
 
+static void Dispatch(void);
+
```
Comment (zevisert) [inline project2/kernel.c]: Can be removed if Dispatch is removed from Kernel_Task_Create

File: project2/kernel.c [hunk 3]
```diff
@@ -161,7 +163,12 @@ void Kernel_Task_Create_At(PD *p, taskfuncptr f) {
 void Kernel_Task_Create() {
     if (Tasks >= MAXTHREAD) {
         /* Too many tasks! */
-        OS_Abort(NO_DEAD_PROCESS);
+        /* OS_Abort(NO_DEAD_PROCESS); */
```
Comment (zevisert) [inline project2/kernel.c]: Should comment here why we don't abort for later

File: project2/kernel.c [hunk 4]
```diff
@@ -412,13 +434,18 @@ void Kernel_Request_MsgSend() {
 
     // Check if process id is valid
     if (!VALID_ID(request_info->msg_to)) {
-        OS_Abort(INVALID_REQ_INFO);
+        /* OS_Abort(INVALID_REQ_INFO); */
```
Comment (zevisert) [inline project2/kernel.c]: Should comment here why we don't abort so this doesn't get re-added

**Pull request merged**